backend/src/apiserver/resource/model_converter.go

```diff
@@ -253,28 +256,31 @@ func runtimeConfigToModelParameters(runtimeConfig *api.PipelineSpec_RuntimeConfi
 		return "", nil
 	}
 	var params []v1alpha1.Parameter
-	for k, v := range runtimeConfig.GetParameters() {
+	// Sort the parameters by names then marshal to string
+	keys := make([]string, 0, len(runtimeConfig.GetParameters()))
+	for key := range runtimeConfig.GetParameters() {
+		keys = append(keys, key)
+	}
+	sort.Strings(keys)
```

Comment:
Since the `Parameters` is a map, the marshalled JSON may be of any random order. I sorted the keys for mainly two reasons:
1. To make comparisons of strings in testing possible
2. Ensure the marshalled JSONs are the same whenever the `Parameters` are the same

Reply:
for #1, as a eng practice, the test shouldn't affect the source code. can you unmarshal the string field and compare without ordering in test code?
for #2, is that necessary for the BE to work properly?

Reply:
for #1, since we are marshalling the dict into a string, the key-value pairs may be marshalled in any random order. This makes the resulting string possibly different every time, thus making testing by comparing the strings directly impossible. An alternative to comparing strings is to unmarshal the strings back into maps, and compare the maps instead. WDYT?

Reply:
yeah i think that's one way to do this correctly.